src/lib/staged_ledger/staged_ledger.ml

```diff
@@ -713,6 +716,9 @@ struct
             (Error.to_string_hum e) ) ;
       Deferred.return (to_staged_ledger_or_error r)
     in
+    Coda_metrics.(
+      Gauge.set Snark_work.scan_state_snark_jobs
+        (Float.of_int (List.length (Scan_state.all_work_pairs_exn scan_state')))) ;
```

Comment:
make this not `_exn` `all_work_pairs` with no exn or catch exn and just skip the metirc if it fails